sql/npc_list.sql

```diff
@@ -29662,7 +29662,7 @@ INSERT INTO `npc_list` VALUES (17764678,'blank','     ',132,59.501,-4.285,32.050
 INSERT INTO `npc_list` VALUES (17764679,'blank','     ',74,127.302,-4.574,97.234,32776,50,50,0,0,64,2,2075,0x0000DA0200000000000000000000000000000000,34,NULL,0);
 INSERT INTO `npc_list` VALUES (17764680,'blank','     ',199,127.790,-4.515,178.663,32769,40,40,0,0,64,2,2075,0x0000DA0200000000000000000000000000000000,34,NULL,0);
 INSERT INTO `npc_list` VALUES (17764681,'blank','     ',135,60.845,-4.635,247.504,32789,50,50,0,0,64,2,2075,0x0000DA0200000000000000000000000000000000,34,NULL,0);
-INSERT INTO `npc_list` VALUES (17764682,'Moogle','Moogle',161,104.823,-5.000,-55.745,1,40,40,0,0,0,2,4194307,0x0000520000000000000000000000000000000000,0,NULL,0);
+INSERT INTO `npc_list` VALUES (17764682,'Egg_Hunt_Moogle','Moogle',161,104.823,-5.000,-55.745,1,40,40,0,0,0,2,4194307,0x0000520000000000000000000000000000000000,0,NULL,0);
```

Comment:
SE moves, deleted, and re-adds these moogles a lot. They will also reuse the same "slot" for different events,

I think we should dynamic entity the seasonal moogles so that they become shift proof.

Reply:
Aside from that concern, this is eggcellent.

Reply:
The name is to ensure the script is kept separate from Mog House or other event Moogles. They're all just called "Moogle" unfortunately. The only way I could 100% identify them was by triggering the cutscene (It turns). Perhaps in the future we will be able to automatically populate IDs like this from the cutscene data.

Reply:
I know, but thats the thing - SE will move them (or delete them) then the ID is invalid. With a dynamic entity we ignore what SE is doing and ***create*** our own moogles ~~with blackjack and hookers~~ and never have to worry that it was moved/deleted

Reply:
This is from Zach's example of a dynamic NPC, it goes in onZoneInitlize in his example:
```lua
    -- Insert NPC into zone
    local horro = zone:insertDynamicEntity({

        -- NPC or MOB
        objtype = xi.objType.NPC,

        -- The name visible to players
        -- NOTE: Even if you plan on making the name invisible, we're using it internally for lookups
        --     : So populate it with something unique-ish even if you aren't going to use it.
        --     : You can then hide the name with entity:hideName(true)
        -- NOTE: This name CAN include spaces and underscores.
        name = "Horro",

        -- You can use regular model ids (See documentation/model_ids.txt, or play around with !costume)
        look = 2430,

        -- You can also use the raw packet look information (as a string), as seen in npc_list and mob_pools
        -- look = "0x0100020500101120003000400050006000700000",

        -- Set the position using in-game x, y and z
        x = 5.000,
        y = 0.000,
        z = 0.000,

        -- Rotation is scaled 0-255, with 0 being East
        rotation = 128,

        -- Overriding widescan is only available to NPCs.
        widescan = 1,

        -- onTrade and onTrigger can be hooked up to your dynamic entity,
        -- just like with regular entities. You can also omit these.
        onTrade = function(player, npc, trade)
            -- NOTE: We have to use getPacketName, because the regular name is modified and being used
            --     : for internal lookups
            player:PrintToPlayer("No, thanks!", 0, npc:getPacketName())
        end,

        -- The entity will not be "triggerable" unless you populate onTrigger
        onTrigger = function(player, npc)
            -- NOTE: We have to use getPacketName, because the regular name is modified and being used
            --     : for internal lookups
            player:PrintToPlayer("Welcome to GM Home!", 0, npc:getPacketName())
        end,
    })

    -- Use the mob object however you like
    -- horro:getID() etc.
    -- We're not doing anything with it, so ignore this object
    utils.unused(horro)

    -- You could also just not capture the object
    -- zone:insertDynamicEntity({ ...end)
```
    
Horro is the NPC from the Dragon Quest crossover event.

Reply:
While this certainly makes sense (And I've used dynamic entities for the decorations where there is no event interaction), there are still a few limitations. For example, the Moogle turns while the player is locked in the cutscene and pivots back afterwards.

We could fake this with `lookAt` but we'd need to only send the update packet to that player, then hook onto `onEventFinish` to reset it. I'm very interested in expanding these kind of functions in the core, as it helps me write convincing custom content. Maybe this is something I will look to pick up in future.

Thanks for your suggestions and kind words.

Reply:
ah right, SE turns the actual NPC ID associated in their event code. amending `lookAt()` and `showText()` (both of them do turning behaviors) has been on my back burner for awhile - some NPCs we currently use it on should only appear to turn for the person interacting with them, and others should turn for _everyone_ but you are right that we'd be doing fakery even with correct event ID launching.

The method used for turning npc's by those functions also has the hilarious unintended side effect I refer to as "bobble head" - they continue to try and follow the player after the interaction, which makes their head bob up and down as you move.

Reply:
I have a work in progress that should solve this facing issue. Should be done pretty soon.

Reply:
`facePlayer()` is now in the upstream